Sync SIPNET user/model docs with current implementation (#380)
PecanProject/sipnet: docs/user-guide/running-sipnet.md

### Configuration File Format

> Each line in the configuration file contains a key-value pair. SIPNET accepts whitespace, `=`, or `:` between the key and value, and `!` starts a comment:

Reviewer: ```suggestion
Each key-value pair is on a separate line separated by a whitespace; `!` starts a comment line:
```

```
KEY = VALUE
```

> Keys are case-insensitive. Non-alphanumeric characters are ignored when matching keys, so hyphens and underscores are interchangeable.

Reviewer: ```suggestion
Keys are case-insensitive, and hyphens and underscores are ignored, so `INPUT_FILE`, `inputfile`, and `iNPUtFiLE` are equivalent.
```

#### Input/Output Keys

| Key                | Value Type | Description                                                                                                       |
|--------------------|------------|-------------------------------------------------------------------------------------------------------------------|
| `FILE_PREFIX`      | string     | Prefix for climate and parameter input files                                                                      |
| `EVENTS_PREFIX`    | string     | Prefix used to derive events input and output filenames                                                           |
| `RESTART_IN`       | string     | Path to checkpoint to resume from                                                                                 |
| `RESTART_OUT`      | string     | Path to checkpoint to write at end of run                                                                         |
| `DEBUG_LOG_PREFIX` | string     | Prefix for debug log files (optional; writes `<prefix>_envi.log`, `<prefix>_fluxes.log`, `<prefix>_trackers.log`) |

Derived paths such as `PARAM_FILE`, `CLIM_FILE`, `OUT_FILE`, and `OUT_CONFIG_FILE` appear in `--dump-config`

> output, but the current implementation recalculates them from `FILE_PREFIX` during startup.

Reviewer: ```suggestion
Derived paths including `PARAM_FILE`, `CLIM_FILE`, `OUT_FILE`, and `OUT_CONFIG_FILE` are derived from `FILE_PREFIX` and appear in `--dump-config` as calculated values.
```

@Alomir is this correct?

Here's an example `sipnet.in` configuration file:

```
! Input files
FILE_PREFIX = my_site

! Model features
EVENTS = 1
GDD = 1
SNOW = 1
NITROGEN_CYCLE = 0
FLOODING = 0

! Output
DO_MAIN_OUTPUT = 1
DO_SINGLE_OUTPUT = 0
PRINT_HEADER = 1

> QUIET = 0

Reviewer: @copilot why did you change this?
